### PR TITLE
DBZ-1385 Document eventType mapping and payload promotion

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: docs/configuration/outbox-event-router.asciidoc [hunk 1]
```diff
@@ -119,7 +119,9 @@ This result was achieved with the link:#default-table-columns[default configurat
 |`table.fields.additional.placement`
 |
 |Table, Envelope
-|Extra fields can be added as part of the event envelope or a message header; the format is a list of colon-delimited pairs or trios when you desire to have aliases, e.g. `id:header,field_name:envelope:alias`. Placement options are `header` and `envelope`
+|Extra fields can be added as part of the event envelope or a message header; the format is a list of colon-delimited pairs or trios when you desire to have aliases, e.g. `id:header,field_name:envelope:alias`. Placement options are `header` and `envelope`. +
```
Comment (renatomefi) [inline docs/configuration/outbox-event-router.asciidoc]:
```suggestion
|Extra fields can be added as part of the event envelope or as message header; the format is a list of colon-delimited pairs or trios when you desire to have aliases, e.g. `id:header,field_name:envelope:alias`. Placement options are `header` and `envelope`. +
```

File: docs/configuration/outbox-event-router.asciidoc [hunk 2]
```diff
@@ -137,7 +139,7 @@ This result was achieved with the link:#default-table-columns[default configurat
 |The default regex to use within the RegexRouter, the default capture will allow to replace the routed field into a new topic name defined in `route.topic.replacement`
 
 |`route.topic.replacement`
-|`outbox.event.${routedByValue}`
+|`outbox.event{zwsp}.${routedByValue}`
```
Comment (renatomefi) [inline docs/configuration/outbox-event-router.asciidoc]: What's `zwsp`?

File: docs/configuration/outbox-event-router.asciidoc [hunk 3]
```diff
@@ -184,10 +186,11 @@ After observing all those pieces we can see what the default configuration does:
 |Becomes the Kafka message key, which is important for keeping ordering within Kafka partitions
 
 |`type`
-|The `type` column becomes part of the message envelope as `eventType`
+|The `type` column becomes either part of the message envelope as `eventType` or message header. +
+See option `table.fields.additional.placement` for more details.
 
 |`payload`
-|The JSON representation of the event itself, becomes part of the message as `payload`
+|The JSON representation of the event itself, becomes either part of the message as `payload` or if other metadata including `eventType` are delivered as headers then the payload becomes the message itself without an encapsulation in an envelope
```
Comment (renatomefi) [inline docs/configuration/outbox-event-router.asciidoc]: Maybe also mention this behavior is attached to the additional fields one?

File: docs/configuration/outbox-event-router.asciidoc (hunk 2 above)
Comment (jpechane) [reply docs/configuration/outbox-event-router.asciidoc]: @renatomefi zero whitespace - allows to break line in this place for formatting

File: docs/configuration/outbox-event-router.asciidoc (hunk 3 above)
Comment (jpechane) [reply docs/configuration/outbox-event-router.asciidoc]: @renatomefi I ma not sure I understand your suggestion, could you please elaborat more or propose the update, thanks!

**Pull request merged**